// @aws-sdk/client-polly/dist-cjs/index.js
'use strict';

var client$1 = require('@aws-sdk/core/client');
var middlewareEventstream = require('@aws-sdk/middleware-eventstream');
var core = require('@smithy/core');
var client = require('@smithy/core/client');
var config = require('@smithy/core/config');
var endpoints = require('@smithy/core/endpoints');
var eventStreams = require('@smithy/core/event-streams');
var protocols = require('@smithy/core/protocols');
var retry = require('@smithy/core/retry');
var schema = require('@smithy/core/schema');
var httpAuthSchemeProvider = require('./auth/httpAuthSchemeProvider');
var runtimeConfig = require('./runtimeConfig');
var schemas_0 = require('./schemas/schemas_0');
var errors = require('./models/errors');
var PollyServiceException = require('./models/PollyServiceException');

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "polly",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(client$1.getAwsRegionExtensionConfiguration(runtimeConfig), client.getDefaultExtensionConfiguration(runtimeConfig), protocols.getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, client$1.resolveAwsRegionExtensionConfiguration(extensionConfiguration), client.resolveDefaultRuntimeConfig(extensionConfiguration), protocols.resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class PollyClient extends client.Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = runtimeConfig.getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = client$1.resolveUserAgentConfig(_config_1);
        const _config_3 = retry.resolveRetryConfig(_config_2);
        const _config_4 = config.resolveRegionConfig(_config_3);
        const _config_5 = client$1.resolveHostHeaderConfig(_config_4);
        const _config_6 = endpoints.resolveEndpointConfig(_config_5);
        const _config_7 = eventStreams.resolveEventStreamSerdeConfig(_config_6);
        const _config_8 = httpAuthSchemeProvider.resolveHttpAuthSchemeConfig(_config_7);
        const _config_9 = middlewareEventstream.resolveEventStreamConfig(_config_8);
        const _config_10 = resolveRuntimeExtensions(_config_9, configuration?.extensions || []);
        this.config = _config_10;
        this.middlewareStack.use(schema.getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(client$1.getUserAgentPlugin(this.config));
        this.middlewareStack.use(retry.getRetryPlugin(this.config));
        this.middlewareStack.use(protocols.getContentLengthPlugin(this.config));
        this.middlewareStack.use(client$1.getHostHeaderPlugin(this.config));
        this.middlewareStack.use(client$1.getLoggerPlugin(this.config));
        this.middlewareStack.use(client$1.getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(core.getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: httpAuthSchemeProvider.defaultPollyHttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new core.DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(core.getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

class DeleteLexiconCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Parrot_v1", "DeleteLexicon", {})
    .n("PollyClient", "DeleteLexiconCommand")
    .sc(schemas_0.DeleteLexicon$)
    .build() {
}

class DescribeVoicesCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Parrot_v1", "DescribeVoices", {})
    .n("PollyClient", "DescribeVoicesCommand")
    .sc(schemas_0.DescribeVoices$)
    .build() {
}

class GetLexiconCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Parrot_v1", "GetLexicon", {})
    .n("PollyClient", "GetLexiconCommand")
    .sc(schemas_0.GetLexicon$)
    .build() {
}

class GetSpeechSynthesisTaskCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Parrot_v1", "GetSpeechSynthesisTask", {})
    .n("PollyClient", "GetSpeechSynthesisTaskCommand")
    .sc(schemas_0.GetSpeechSynthesisTask$)
    .build() {
}

class ListLexiconsCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Parrot_v1", "ListLexicons", {})
    .n("PollyClient", "ListLexiconsCommand")
    .sc(schemas_0.ListLexicons$)
    .build() {
}

class ListSpeechSynthesisTasksCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Parrot_v1", "ListSpeechSynthesisTasks", {})
    .n("PollyClient", "ListSpeechSynthesisTasksCommand")
    .sc(schemas_0.ListSpeechSynthesisTasks$)
    .build() {
}

class PutLexiconCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Parrot_v1", "PutLexicon", {})
    .n("PollyClient", "PutLexiconCommand")
    .sc(schemas_0.PutLexicon$)
    .build() {
}

class StartSpeechSynthesisStreamCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [
        endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions()),
        middlewareEventstream.getEventStreamPlugin(config),
    ];
})
    .s("Parrot_v1", "StartSpeechSynthesisStream", {
    eventStream: {
        input: true,
        output: true,
    },
})
    .n("PollyClient", "StartSpeechSynthesisStreamCommand")
    .sc(schemas_0.StartSpeechSynthesisStream$)
    .build() {
}

class StartSpeechSynthesisTaskCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Parrot_v1", "StartSpeechSynthesisTask", {})
    .n("PollyClient", "StartSpeechSynthesisTaskCommand")
    .sc(schemas_0.StartSpeechSynthesisTask$)
    .build() {
}

class SynthesizeSpeechCommand extends client.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [endpoints.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Parrot_v1", "SynthesizeSpeech", {})
    .n("PollyClient", "SynthesizeSpeechCommand")
    .sc(schemas_0.SynthesizeSpeech$)
    .build() {
}

const paginateListSpeechSynthesisTasks = core.createPaginator(PollyClient, ListSpeechSynthesisTasksCommand, "NextToken", "NextToken", "MaxResults");

const commands = {
    DeleteLexiconCommand,
    DescribeVoicesCommand,
    GetLexiconCommand,
    GetSpeechSynthesisTaskCommand,
    ListLexiconsCommand,
    ListSpeechSynthesisTasksCommand,
    PutLexiconCommand,
    StartSpeechSynthesisStreamCommand,
    StartSpeechSynthesisTaskCommand,
    SynthesizeSpeechCommand,
};
const paginators = {
    paginateListSpeechSynthesisTasks,
};
class Polly extends PollyClient {
}
client.createAggregatedClient(commands, Polly, { paginators });

const Engine = {
    GENERATIVE: "generative",
    LONG_FORM: "long-form",
    NEURAL: "neural",
    STANDARD: "standard",
};
const LanguageCode = {
    ar_AE: "ar-AE",
    arb: "arb",
    ca_ES: "ca-ES",
    cmn_CN: "cmn-CN",
    cs_CZ: "cs-CZ",
    cy_GB: "cy-GB",
    da_DK: "da-DK",
    de_AT: "de-AT",
    de_CH: "de-CH",
    de_DE: "de-DE",
    en_AU: "en-AU",
    en_GB: "en-GB",
    en_GB_WLS: "en-GB-WLS",
    en_IE: "en-IE",
    en_IN: "en-IN",
    en_NZ: "en-NZ",
    en_SG: "en-SG",
    en_US: "en-US",
    en_ZA: "en-ZA",
    es_ES: "es-ES",
    es_MX: "es-MX",
    es_US: "es-US",
    fi_FI: "fi-FI",
    fr_BE: "fr-BE",
    fr_CA: "fr-CA",
    fr_FR: "fr-FR",
    hi_IN: "hi-IN",
    is_IS: "is-IS",
    it_IT: "it-IT",
    ja_JP: "ja-JP",
    ko_KR: "ko-KR",
    nb_NO: "nb-NO",
    nl_BE: "nl-BE",
    nl_NL: "nl-NL",
    pl_PL: "pl-PL",
    pt_BR: "pt-BR",
    pt_PT: "pt-PT",
    ro_RO: "ro-RO",
    ru_RU: "ru-RU",
    sv_SE: "sv-SE",
    tr_TR: "tr-TR",
    yue_CN: "yue-CN",
};
const Gender = {
    Female: "Female",
    Male: "Male",
};
const VoiceId = {
    Aditi: "Aditi",
    Adriano: "Adriano",
    Ambre: "Ambre",
    Amy: "Amy",
    Andres: "Andres",
    Aria: "Aria",
    Arlet: "Arlet",
    Arthur: "Arthur",
    Astrid: "Astrid",
    Ayanda: "Ayanda",
    Beatrice: "Beatrice",
    Bianca: "Bianca",
    Brian: "Brian",
    Burcu: "Burcu",
    Camila: "Camila",
    Carla: "Carla",
    Carmen: "Carmen",
    Celine: "Celine",
    Chantal: "Chantal",
    Conchita: "Conchita",
    Cristiano: "Cristiano",
    Daniel: "Daniel",
    Danielle: "Danielle",
    Dora: "Dora",
    Elin: "Elin",
    Emma: "Emma",
    Enrique: "Enrique",
    Ewa: "Ewa",
    Filiz: "Filiz",
    Florian: "Florian",
    Gabrielle: "Gabrielle",
    Geraint: "Geraint",
    Giorgio: "Giorgio",
    Gregory: "Gregory",
    Gwyneth: "Gwyneth",
    Hala: "Hala",
    Hannah: "Hannah",
    Hans: "Hans",
    Hiujin: "Hiujin",
    Ida: "Ida",
    Ines: "Ines",
    Isabelle: "Isabelle",
    Ivy: "Ivy",
    Jacek: "Jacek",
    Jan: "Jan",
    Jasmine: "Jasmine",
    Jihye: "Jihye",
    Jitka: "Jitka",
    Joanna: "Joanna",
    Joey: "Joey",
    Justin: "Justin",
    Kajal: "Kajal",
    Karl: "Karl",
    Kazuha: "Kazuha",
    Kendra: "Kendra",
    Kevin: "Kevin",
    Kimberly: "Kimberly",
    Laura: "Laura",
    Lea: "Lea",
    Lennart: "Lennart",
    Liam: "Liam",
    Lisa: "Lisa",
    Liv: "Liv",
    Lorenzo: "Lorenzo",
    Lotte: "Lotte",
    Lucia: "Lucia",
    Lupe: "Lupe",
    Mads: "Mads",
    Maja: "Maja",
    Marlene: "Marlene",
    Mathieu: "Mathieu",
    Matthew: "Matthew",
    Maxim: "Maxim",
    Mia: "Mia",
    Miguel: "Miguel",
    Mizuki: "Mizuki",
    Naja: "Naja",
    Niamh: "Niamh",
    Nicole: "Nicole",
    Ola: "Ola",
    Olivia: "Olivia",
    Pedro: "Pedro",
    Penelope: "Penelope",
    Raveena: "Raveena",
    Remi: "Remi",
    Ricardo: "Ricardo",
    Ruben: "Ruben",
    Russell: "Russell",
    Ruth: "Ruth",
    Sabrina: "Sabrina",
    Salli: "Salli",
    Seoyeon: "Seoyeon",
    Sergio: "Sergio",
    Sofie: "Sofie",
    Stephen: "Stephen",
    Suvi: "Suvi",
    Takumi: "Takumi",
    Tatyana: "Tatyana",
    Thiago: "Thiago",
    Tiffany: "Tiffany",
    Tomoko: "Tomoko",
    Vicki: "Vicki",
    Vitoria: "Vitoria",
    Zayd: "Zayd",
    Zeina: "Zeina",
    Zhiyu: "Zhiyu",
};
const OutputFormat = {
    ALAW: "alaw",
    JSON: "json",
    MP3: "mp3",
    MULAW: "mulaw",
    OGG_OPUS: "ogg_opus",
    OGG_VORBIS: "ogg_vorbis",
    PCM: "pcm",
};
const SpeechMarkType = {
    SENTENCE: "sentence",
    SSML: "ssml",
    VISEME: "viseme",
    WORD: "word",
};
const TaskStatus = {
    COMPLETED: "completed",
    FAILED: "failed",
    IN_PROGRESS: "inProgress",
    SCHEDULED: "scheduled",
};
const TextType = {
    SSML: "ssml",
    TEXT: "text",
};
const QuotaCode = {
    INPUT_STREAM_INBOUND_EVENT_TIMEOUT: "input-stream-inbound-event-timeout",
    INPUT_STREAM_TIMEOUT: "input-stream-timeout",
};
const ServiceCode = {
    POLLY: "polly",
};
const ValidationExceptionReason = {
    FIELD_VALIDATION_FAILED: "fieldValidationFailed",
    INVALID_INBOUND_EVENT: "invalidInboundEvent",
    OTHER: "other",
    UNSUPPORTED_OPERATION: "unsupportedOperation",
};

exports.$Command = client.Command;
exports.__Client = client.Client;
exports.PollyServiceException = PollyServiceException.PollyServiceException;
exports.DeleteLexiconCommand = DeleteLexiconCommand;
exports.DescribeVoicesCommand = DescribeVoicesCommand;
exports.Engine = Engine;
exports.Gender = Gender;
exports.GetLexiconCommand = GetLexiconCommand;
exports.GetSpeechSynthesisTaskCommand = GetSpeechSynthesisTaskCommand;
exports.LanguageCode = LanguageCode;
exports.ListLexiconsCommand = ListLexiconsCommand;
exports.ListSpeechSynthesisTasksCommand = ListSpeechSynthesisTasksCommand;
exports.OutputFormat = OutputFormat;
exports.Polly = Polly;
exports.PollyClient = PollyClient;
exports.PutLexiconCommand = PutLexiconCommand;
exports.QuotaCode = QuotaCode;
exports.ServiceCode = ServiceCode;
exports.SpeechMarkType = SpeechMarkType;
exports.StartSpeechSynthesisStreamCommand = StartSpeechSynthesisStreamCommand;
exports.StartSpeechSynthesisTaskCommand = StartSpeechSynthesisTaskCommand;
exports.SynthesizeSpeechCommand = SynthesizeSpeechCommand;
exports.TaskStatus = TaskStatus;
exports.TextType = TextType;
exports.ValidationExceptionReason = ValidationExceptionReason;
exports.VoiceId = VoiceId;
exports.paginateListSpeechSynthesisTasks = paginateListSpeechSynthesisTasks;
Object.prototype.hasOwnProperty.call(schemas_0, '__proto__') &&
    !Object.prototype.hasOwnProperty.call(exports, '__proto__') &&
    Object.defineProperty(exports, '__proto__', {
        enumerable: true,
        value: schemas_0['__proto__']
    });

Object.keys(schemas_0).forEach(function (k) {
    if (k !== 'default' && !Object.prototype.hasOwnProperty.call(exports, k)) exports[k] = schemas_0[k];
});
Object.prototype.hasOwnProperty.call(errors, '__proto__') &&
    !Object.prototype.hasOwnProperty.call(exports, '__proto__') &&
    Object.defineProperty(exports, '__proto__', {
        enumerable: true,
        value: errors['__proto__']
    });

Object.keys(errors).forEach(function (k) {
    if (k !== 'default' && !Object.prototype.hasOwnProperty.call(exports, k)) exports[k] = errors[k];
});
